src/Dialog/Content/index.tsx

```diff
@@ -141,7 +141,8 @@ const Content = React.forwardRef<ContentRef, ContentProps>((props, ref) => {
       {({ className: motionClassName, style: motionStyle }, motionRef) => (
         <div
           key="dialog-element"
-          role="document"
+          role="dialog"
```

Comment:
Can you explain why you are recommending the addition of an attribute whose purpose is purely for the benefit of screen reader users to solve a problem with how your specific app has decided to define RTL selectors?

Reply:
Your understanding of those aria attributes is completely incorrect.

Reply:
How are those aria attributes not helpful? Please help me understand.

Reply:
@hedaukartik Sure. ARIA attributes should ONLY be used for accessibility purposes for screen reader users. It sound like what you're looking for is simply a way to identify a specific element in the DOM for your automated UI tests. I'm sure you could search for an article that talks about the best way to identify a selector for tests, but the best practice is to use an attribute that is there specifically for tests. Here's an example:

```
<div class="my-app">
  <h1>My Header</h1>
  <div class="content">
    <a href="#" aria-label="Home">
        <img src="/icon" />
    </a>
  </div>
</div>
```

If you are writing a test that needs to let's say check the link of the anchor, what selector could you use?

`a`? That obviously wouldn't work out to well since the second another anchor gets added to the page, you're in trouble.
`.content a`? Better, but classes are used for CSS so what happens if/when you change the class `.content`?
`a[aria-label="Home"]`? ARIA is used purely for screen reader users so the second Content Management decides to change the aria-label, your test breaks.

The idea situation is if you can put a test-specific attribute on things you need to test.  Again using the example above and trying to select the anchor, something like this: `<a href="#" aria-label="Home" data-testid="home-link">`.  Now in my test, I can always use `[data-testid="home-link"]` to select that element.

The complication here is that, when we're talking about third-party libraries, you don't have control of all the markup to be able to set a test-specific identifier. So you have to do your best. But don't use ARIA as a selector for a test. 

Hope this helps.

Reply:
@mellis481 You are misunderstanding the intent of Testing Library. It intentionally selects elements by role, not by IDs. You can read a short version of its manifesto here: https://testing-library.com/docs/react-testing-library/intro/#this-solution

This is not strictly an ARIA issue - there is just no better way to label an element with a role of `dialog` than by using `aria-labelledby`. Please see this MDN article on how to label dialogs: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role

I don't think there's any issue with changing the element that has the role of `dialog`. The issue with this PR is that `aria-labelledby` should have been moved along with the `role`, and it was not.

Reply:
It looks like this PR fixes the oversight: https://github.com/react-component/dialog/pull/290

Reply:
@JeffreyATW “Oversight” is a bit of an over-exaggeration. The PR that broke your tests was created to improve the lives of screen readers after a professional 3rd party accessibility auditor reviewed the AntD library.

I hope that it’s clear that the role was previously on something that was not actually the modal (it was on the background if I recall correctly from @mellis481 ’s PR). So that change was necessary to improve the relationship with the screen readers and also to get this component to pass automated a11y auditing tools.

If tests need to change, they need to change so we can support users who need our help.

If #290 allows a workaround without creating new a11y problems then that’s great. But from my perspective, the test code is incorrect not the component itself. Let’s make sure that #290 doesn’t create extra hardship for screen reader users.

Reply:
If you look at the [page you linked in your first post](https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html), you'll see that `aria-labelledby` should be applied to the same element as `role="dialog"`.

I agree that `role="dialog"` was applied to the wrong element prior to this PR. But `aria-labelledby` should have been moved along with it. 

Please see this comment for my findings when using a screen reader: https://github.com/react-component/dialog/pull/290#issuecomment-1126244246

Reply:
Thank you for confirming that #290 doesn’t create hardship. I’m pleased to know that it actually improves the accessibility. Thank you so much. It seems like the fix PR has the correct HTML as you mentioned. I commented asking Zombie to consider it.

Reply:
@JeffreyATW It sounds like I did not correctly interpret the message @hedaukartik was trying to communicate. I'm glad that this PR and #290 that has been created will get the modal in a much better position a11y-wise.